### DISCUSSION TITLE
Each time stream goes through ome layer, the first 2 seconds of the video are missing.

### DISCUSSION THREAD

**Discussion by holynls** [OvenMediaLabs/OvenMediaEngine#1681] [Help] [answered]
I am currently building a stream service with three ome layers: a receiver that receives streams via the SRT protocol at the front end for each region, a worker that transcodes the video and audio received from the receiver into HLS and LLHLS formats and also by-pass them to the edge, and an edge layer that serves the video.

Recently, I discovered that the beginning of the transcoded HLS videos was missing. Upon further investigation, I found that approximately 2 seconds of the stream appear to be missing each time it passes through a layer.

For example, the video transcoded from receiver to worker has about 4 seconds missing from the start. Additionally, when viewing the video after it passes from receiver to worker to edge, about 6 seconds are missing from the beginning.

Has anyone experienced this issue before or have any hints on how to resolve it?

**Comment by bchah** [comment]
I believe HLS segments need to begin with a keyframe. I've seen that behavior before where OME will create a longer-than-expected segment because the keyframe interval is off by even 1ms. This could be related to the behavior you're seeing.

**Comment by getroot** [comment, marked as answer]
I think ovt's request is made after origin's stream is created, so it won't receive the origin's first frame. Then edge can receive from the next keyframe, so it will miss that much.

**Reply by holynls** [reply]
I was streaming with the keyframe interval set to 2 seconds, and now I think I understand why this happened. It makes sense. Thank you so much for your help.
